src/TestFramework/Coverage/XmlReport/JUnitTestCaseSorter.php

```diff
@@ -63,31 +73,25 @@ static function (CoverageLineData $a, CoverageLineData $b) {
             }
         );
 
-        return array_map(
-            static function (CoverageLineData $coverageLineData): string {
-                $testFilePath = $coverageLineData->testFilePath;
-                Assert::string($testFilePath);
-
-                return $testFilePath;
-            },
-            $uniqueCoverageTests
-        );
+        foreach ($uniqueCoverageTests as $coverageLineData) {
+            yield $coverageLineData->testFilePath;
+        }
     }
 
     /**
      * @param CoverageLineData[] $coverageTestCases
      *
      * @return CoverageLineData[]
      */
-    private function uniqueByTestFile(array $coverageTestCases): array
+    private function uniqueByTestFile(iterable $coverageTestCases): array
```

Comment:
AFAICS it's `array` here, it's coming from `getUniqueSortedFileNames(array $coverageTestCases)` so there is no need to expand the type

Reply:
Fair. Let's not expand.

Reply:
```suggestion
    private function uniqueByTestFile(array $coverageTestCases): array
```